arrow/src/main/java/org/apache/iceberg/arrow/vectorized/NullabilityHolder.java

```diff
@@ -75,6 +77,7 @@ public int numNulls() {
   }
 
   public void reset() {
+    // TODO: do we have reset the values too?
```

Comment:
There are cases when we reuse the nullability holder and call `reset` and `setNotNulls(0, numValsToRead)`. It looks like we still have to invalidate the underlying values?

Reply:
Again I think the assumption of this class is that you never call a "set" method more than once on an index and that you call it on every index.

Reply:
It is not documented, though. @samarthjain, could you confirm our understanding?

Reply:
That's correct, @aokolnychyi . When I wrote this, there was inherent assumption that the user of this class will be calling `setNull(index)` and `setNotNull(index)` for every row in the batch which is why reset() didn't reinitialize or assign a new `byte[]` for `isNull`. See the comment at the top the class
```
/**
 * Instances of this class simply track whether a value at an index is null.
 * For simplicity and performance, it is expected that various setter methods
 * {@link #setNull(int)}, {@link #setNulls(int, int)}, {@link #setNotNull(int)}
 * and {@link #setNotNulls(int, int)} are invoked with monotonically
 * increasing values for the index parameter.
 */
 ```
Although, looking at this code now, I agree with @RussellSpitzer that we could simply reinitialize the `isNull` array again.

Reply:
I think it is fine the way it is. We have to set the value for each index. Otherwise, no guarantee it is going to be correct.

Reply:
I think this can be resolved now.